backend/server/routers/movies.py

```diff
@@ -22,6 +29,35 @@
 @router.get("/movies")
 async def movies(user_id: str = Depends(auth)):
     results = await DB.query(
-        "SELECT `id`, `title`, `release_date` as `showings_start`, `poster_url` FROM `MovieShowing`"
+        "SELECT `id`, `title`, `release_date` AS `showings_start`, `poster_url` FROM `MovieShowing`"
     )
+    # Check for existing reviews for each movie in the results for this user
+    for movie in results:
+        review_key = f"{user_id}:{movie['id']}"
+        movie["user_review"] = reviews_store.get(review_key)
     return results
+
+@router.post("/movie/review", status_code=200)
+async def create_movie_review(
+    review: CreateMovieReviewRequest,
+    user_id: str = Depends(auth),
+):
+    # Create a unique key to identify a review by this user for this movie
```

Comment:
The logic here is correct, it just needs to be done with the database, not an in-memory store.